src/v/net/connection.cc

```diff
@@ -53,6 +54,7 @@ bool is_reconnect_error(const std::system_error& e) {
         case ECONNABORTED:
         case EAGAIN:
         case EPIPE:
+        case EHOSTUNREACH:
```

Comment:
I wonder if there are any other error codes we could opportunistically add here? E.g.:

1. `ESHUTDOWN`
2. `EHOSTDOWN`

Reply:
I think `ESHUTDOWN` is where we shutdown the socket (https://linux.die.net/man/3/shutdown).  So probably an artifact of shutting down and not a retry.  Thoughts?

I think the other three can be added.

Reply:
We probably shut down the socket on all manner of alternative errors. At initial glance, I didn't see an abort source which would be more explicit about shutdown.